File: Programming Languages/4 languages in 2 styles/OO/Python/Python OO/DocumentProcessorBuilder.py
import LineFilterer, CaseConverter, WordFinder, NonABCFilterer, WordCounter


class DocumentProcessorBuilder:

    def set_line_filterer(self):
        self.line_filterer = LineFilterer.LineFilterer()

    def set_case_converter(self):
        self.case_converter = CaseConverter.CaseConverter()


    def set_word_finder(self):
        self.word_finder = WordFinder.WordFinder()

    def set_non_abc_filterer(self):
        self.non_abc_filterer = NonABCFilterer.NonABCFilterer()

    def set_word_counter(self):
        self.word_counter = WordCounter.WordCounter()

File: Programming Languages/4 languages in 2 styles/OO/Javascript/Javascript OO/DocumentProcessorBuilder.mjs
import {LineFilterer} from "./Interfaces/LineFilterer";
import {CaseConverter} from "./Interfaces/CaseConverter";
import {NonABCFilterer} from "./Interfaces/NonABCFilterer";
import {WordCounter} from "./Interfaces/WordCounter";
import {WordFinder} from "./Interfaces/WordFinder";

export class DocumentProcessorBuilder {


    constructor() {
       // Array;
    }

    setLineFilterer() {
        this.LineFilterer = new LineFilterer();
    }

    setCaseConverter() {
        this.CaseConverter = new CaseConverter();
    }

    setNonABCFilterer() {
        this.NonABCFilterer = new NonABCFilterer();
    }

    setWordCounter() {
        this.WordCounter = new WordCounter();
    }

    setWordFinder() {
        this.WordFinder = new WordFinder();
    }


}
